src/trio_monitor/_tests/test_monitor.py

```diff
@@ -0,0 +1,70 @@
+import trio
+import trio_monitor
+import pytest
+
+
+def run_with_monitor(async_fn, monitor):
+    trio.run(
+        async_fn,
+        instruments=[monitor],
+        clock=trio.testing.MockClock(autojump_threshold=0),
+    )
+
+
+def test_it_does_not_crash():
+    async def main():
+        async with trio.open_nursery() as nursery:
+            nursery.start_soon(trio.sleep, 1)
+
+    run_with_monitor(main, trio_monitor.Monitor())
+
+
+def test_it_can_start_a_server():
```

Comment:
I think this is the wrong way around it imo. Testing the user-focused cli output is probably not a good idea; there should be some underlying protocol that is more easily tested and then the evilness of turning the RPC into user-focused output can be entirely isolated.

Reply:
Agreed, except that (for now) the only exposed output is this console...

(One level lower would be calling methods on Monitor which is a decent idea so maybe I should do that.)

Reply:
I tried out calling methods on `Monitor` and it's a bit... not great. But at least I made a test for the core functionality of monitoring a program.